### PR TITLE
sklearn and crf params from config

### PR DESCRIPTION
**Pull request by IanRogers-LabShift**
**Proposed changes**: 
- Expose some hardwired parameters so they are settable from config in pipeline items: "intent_classifier_sklearn" "ner_crf".
- Add a `from_dict` class-method to `RasaNLUConfig` so test code conforms to mypy

**Status**:
- [x] ready for code review
- [ ] there are tests for the functionality
- [ ] documentation updated
- [ ] changelog updated

### CODE REVIEW

**Comment by wrathagom** [comment]
@tmbo would you want any specific tests for this?

@IanRogers-LabShift are you working on updating docs as well?

**Comment by IanRogers-LabShift** [comment]
@wrathagom I hadn't planned on writing any docs as such - looking at the other code there seems to be a "use the source, Luke" convention at the moment :-)

**Comment by wrathagom** [comment]
@IanRogers-LabShift I am referring to the read the docs http://rasa-nlu.readthedocs.io/en/latest/ since I'm just a contributor here and not an owner I can say that I find the docs to be quite nice.

Specifically configuration parameters and there defaults go on this page. http://rasa-nlu.readthedocs.io/en/latest/config.html

File: rasa_nlu/classifiers/sklearn_intent_classifier.py [hunk 1]
```diff
@@ -93,8 +93,11 @@ def train(self, training_data, config, **kwargs):
             y = self.transform_labels_str2num(labels)
             X = np.stack([example.get("text_features") for example in training_data.intent_examples])
 
+            config_dict = config.as_dict()
```
Comment (tmbo) [inline rasa_nlu/classifiers/sklearn_intent_classifier.py]: I think you should be able to do `config.get("sklearn_c", xyz)` directly as the Config class should be a wrapper around a dict (so no need for the `.as_dict()`).

File: rasa_nlu/config.py [hunk 2]
```diff
@@ -87,6 +87,12 @@ def __init__(self, filename=None, env_vars=None, cmdline_args=None):
         for key, value in self.items():
             setattr(self, key, value)
 
+    @classmethod
+    def from_dict(cls, dict):
+        config = cls.__new__(cls)
```
Comment (tmbo) [inline rasa_nlu/config.py]: shouldn't `cls()` work as well?

**Comment by wrathagom** [comment]
I restarted the one that failed, looked like a memory problem not related to the PR.

**Comment by tmbo** [comment]
Great work, looks really good. Thanks a lot for the work you have put into the PR. 

One minor last thing: Could you please rename the configuration parameters to be the same as the component name the configuration is used for (e.g. `entity_crf_train ` to `ner_crf`)? After that we can merge :+1:

**Comment by tmbo** [comment]
Thanks so much for your contribution @IanRogers-LabShift very happy to merge this :)

**Comment by IanRogers-LabShift** [comment]
@tmbo thanks. NB. @nikitakarandej and @Ayoubb94 did the work :-) , I'm just the team-lead on our project :-)

**Pull request merged**